portfolio/src/main/webapp/script.js

```diff
@@ -36,23 +37,33 @@ function currentSlide(n) {
     slideIndex = n
     showSlides();
 }
-
+//function that hides any slides that aren't at the slideIndex in the list of mySlides,
+//and shows the selected slide with the correct size according to the showLargeImages variable
 function showSlides() {
   const slides = document.getElementsByClassName("mySlides");
   const dots = document.getElementsByClassName("dot");
 
+  //hide any unselected slides
   for(slide of slides)
     slide.style.display = 'none';
 
-  slideIndex = slideIndex % slides.length; 
-  
+  //ensure slide index is not greater than the max index
+  slideIndex = slideIndex % slides.length;
+
+  //resize photo according to the showLargeImages global variable
+  resizePhotoAuto(slides[slideIndex].getElementsByTagName('img')[0]);
+
+  //de-highlight all dots
   for(dot of dots)
     dot.className = dot.className.replace(" active","");
+
+  //Make current slide show with block formatting
+  slides[slideIndex].style.display = "block";
   
-  slides[slideIndex].style.display = "block";  
+  //highlight dot that represents selected image on screen
   dots[slideIndex].className += " active";
```

Comment:
When commenting code, try to focus on _why_ and not _what_. As a general rule, assume whoever is reading your code (which could be your code reviewer, a stranger on the internet, or yourself in 6 months) knows whatever language you're working in at least as well as you do right now (if not better).

If we assume that, explaining "what" a line of code does in English isn't very helpful. I think this applies to all of your comments between 46 and 63. Ideally, most code should be clear enough that a comment is not necessary.

In some cases, it may be useful to say _why_ a particular piece of code exists. For example, on line 46, you wrote "hide any unselected slides". I would change that comment to something like "Hide all of the slides to get to a clean state in preparation for showing a single slide." If you want to make a small change to this code in 6 months, hopefully that comment will jog your memory on what you were thinking about when you originally wrote the code.

(Pedantically, lines 47 and 48 hide all of the slides and not just the unselected ones)